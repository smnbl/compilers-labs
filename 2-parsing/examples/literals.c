int literals()
{
    int x = 42;
    float y = 42.5;
    string z = "forty-two";
}
