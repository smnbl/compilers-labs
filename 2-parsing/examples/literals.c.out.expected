└── Program
    └── FuncDecl: returnType = 'int', name = 'literals'
        └── CompoundStmt
            ├── VarDecl: type = 'int', name = 'x'
            │   └── IntLiteral: value = '42'
            ├── VarDecl: type = 'float', name = 'y'
            │   └── FloatLiteral: value = '42.5'
            └── VarDecl: type = 'string', name = 'z'
                └── StringLiteral: value = 'forty-two'
